### PR TITLE
Update docs for goto-programs

### PR DESCRIPTION
**Pull request by smowton**

### CODE REVIEW

**Review by owen-mc-diffblue — APPROVED** [review]

File: src/goto-programs/README.md [hunk 1]
```diff
@@ -8,100 +8,59 @@
...
+The `cbmc` option `--show-goto-programs` is often a good starting point
 as it outputs goto-programs in a human readable form. However there are
 a few things to be aware of. Functions have an internal name (for
 example `c::f00`) and a ‘pretty name’ (for example `f00`) and which is
 used depends on whether it is internal or being presented to the user.
-The `main` method is the ‘logical’ main which is not necessarily the
-main method from the code. In the output `NONDET` is use to represent a
-nondeterministic assignment to a variable. Likewise `IF` as a beautified
+In the output `NONDET` is use to represent a nondeterministic assignment to a
+variable. Likewise `IF` as a beautified
```
Comment (tautschnig) [inline src/goto-programs/README.md]: I know that's just re-formatting of prior text, but none of these two sentences fully parse for me.

File: src/goto-programs/README.md [hunk 2]
```diff
@@ -8,100 +8,59 @@
...
 `GOTO` instruction where the guard expression is used as the condition.
-`RETURN` instructions may be dropped if they precede an `END_FUNCTION`
-instruction. The comment lines are generated from the `locationt` field
+The comment lines are generated from the `source_locationt` field
```
Comment (tautschnig) [inline src/goto-programs/README.md]: That field is named `source_location` (without a t)

**Review by tautschnig — APPROVED** [review]

**Comment by smowton** [comment]
@tautschnig fixed

**Comment by smowton** [comment]
@peterschrammel ready I think

**Pull request merged**